R/interpolate.R

```diff
@@ -63,12 +65,22 @@ interpolate_package <- function(
   ...,
   .envir = parent.frame()
 ) {
-  path <- system.file("prompts", path, package = package)
-  interpolate_file(path, ..., .envir = .envir)
+  check_string(path)
+
+  path_pkg <- system.file("prompts", path, package = package)
+
+  if (!nzchar(path_pkg)) {
+    cli::cli_abort(c(
+      "{.pkg {package}} does not have {.val {path}} in its {.field prompts/} directory.",
+      "i" = 'Run {.run dir(system.file("prompts", package = "{package}"))} to see available prompts.'
+    ))
+  }
+
+  interpolate_file(path_pkg, ..., .envir = .envir)
 }
 
 read_file <- function(path) {
-  file_contents <- readChar(path, file.size(path))
+  readChar(path, file.size(path))
```

Comment:
Given the previous uninformative error, I'd be tempted to check that `path` exists here too.

Reply:
Yeah, I was also tempted, but the default error through `interpolate_file()` makes more sense. (It could definitely be improved, of course)

```r
ellmer::interpolate_file("foo")
#> Warning message:
#> In file(con, "rb") : cannot open file 'foo': No such file or directory
#> Error in `file()`:
#> ! cannot open the connection
```

I didn't add a custom error because I thought it's a lot more likely to be clear why the call failed or that users would have encountered a "cannot open the connection" error before.

Still, happy to improve the error message here too if you'd like!